lib/StackAnalysis/StackAnalysis.cpp

```diff
@@ -65,6 +88,232 @@ static opt<std::string> ABIAnalysisOutputPath("abi-analysis-output",
+  auto *Call = GCBI::getFunctionCall(Term);
+
+  // If the successor is null, we are dealing with an indirect call.
+  auto *Next = GCBI::getSuccessor(Term);
+
+  // What are the registers clobbered by the callee?
+  const auto &ClobberedRegs = Oracle.getRegistersClobbered(Next);
+
+  // What is the function type of the callee?
+  FunctionKind Ty = Oracle.getFunctionType(Next);
+
+  switch (Ty) {
+  case FunctionKind::Regular: {
+    BranchInst *Br = BranchInst::Create(GCBI::getFallthrough(Term));
+    ReplaceInstWithInst(Term, Br);
+
+    // Prevent the store instructions from being optimized out by storing
+    // the rval of a call to an opaque function into the clobbered registers.
+    OFPClobbering.addFnAttribute(Attribute::ReadOnly);
+    OFPClobbering.addFnAttribute(Attribute::NoUnwind);
+
+    auto *OpqFunc = OFPClobbering.get(BB->getParent()->getName(),
+                                      Type::getInt64Ty(Context),
+                                      {},
+                                      "opq_func_for_clobbering");
+    for (GlobalVariable *Reg : ClobberedRegs)
+      new StoreInst(CallInst::Create(OpqFunc, "", Br), Reg, Br);
+
+    break;
+  }
+
+  case FunctionKind::NoReturn: {
+    ReturnInst *Ret = ReturnInst::Create(Context);
+    ReplaceInstWithInst(Term, Ret);
+    break;
+  }
+
+  case FunctionKind::Fake: {
+    BranchInst *Br = BranchInst::Create(GCBI::getFallthrough(Term));
+
+    // Get the fake function by its entry basic block.
+    Function *FakeFunction = Oracle.getFakeFunction(Next);
+
+    // Must have already been analyzed.
+    revng_assert(FakeFunction != nullptr);
+
+    // If possible, inline the fake function.
+    auto *CI = CallInst::Create(FakeFunction, "", Term);
+    InlineFunctionInfo IFI;
+    bool Status = InlineFunction(llvm::CallSite(CI),
+                                 IFI,
+                                 nullptr,
+                                 true);
+
+    // Still need to replace the successor with the fall-through.
+    ReplaceInstWithInst(Term, Br);
+    break;
+  }
+
+  default:
+    revng_abort();
+  }
+
+  Call->eraseFromParent();
+}
+
+template <class FunctionOracle>
+llvm::Function *CFEPAnalyzer<FunctionOracle>::createDisposableFunction(llvm::BasicBlock *Entry) {
+  using namespace llvm;
+  Function *Root = Entry->getParent();
+
+  OnceQueue<BasicBlock *> Queue;
+  std::vector<BasicBlock *> BlocksToCloneVec;
+  Queue.insert(Entry);
+
+  while (!Queue.empty()) {
+    BasicBlock *Current = Queue.pop();
+
+    BlocksToCloneVec.emplace_back(Current);
+
+    if (GCBI::isFunctionCall(Current)) {
+      auto *Succ = GCBI::getFallthrough(Current);
+      revng_assert(Succ != nullptr);
+      Queue.insert(Succ);
+      continue;
+    }
+
+    for (auto *Succ : successors(Current)) {
+      if (!GCBI::isPartOfRootDispatcher(Succ))
+        Queue.insert(Succ);
+    }
+  }
+
+  ValueToValueMapTy VMap;
+  SmallVector<BasicBlock *, 8> BlocksToExtractVec;
```

Comment:
Do not put reference to the type in the variable name. [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation) is early 2000s.

Reply:
Gotcha.